xontrib/vox.py

```diff
@@ -75,121 +63,113 @@ def parser():
-            default=False,
-            action="store_true",
-            dest="activate",
-            help="Activate the newly created virtual environment.",
-        )
 
-        activate = subparsers.add_parser(
-            "activate", aliases=["workon", "enter"], help="Activate virtual environment"
-        )
-        activate.add_argument(
-            "name",
-            metavar="ENV",
-            help=(
-                "The environment to activate. ENV can be "
-                "either a name from the venvs shown by vox"
-                "list or the path to an arbitrary venv"
-            ),
-        )
-        deactivate = subparsers.add_parser(
-            "deactivate",
-            aliases=["exit"],
-            help="Deactivate current virtual environment",
-        )
-        deactivate.add_argument(
-            "--remove",
-            dest="remove",
-            default=False,
-            action="store_true",
-            help="Remove the virtual environment after leaving it.",
-        )
-        subparsers.add_parser(
-            "list",
-            aliases=["ls"],
-            help=("List environments available in " "$VIRTUALENV_HOME"),
-        )
-        remove = subparsers.add_parser(
-            "remove", aliases=["rm", "delete", "del"], help="Remove virtual environment"
-        )
-        remove.add_argument(
-            "names",
-            metavar="ENV",
-            nargs="+",
-            help=(
-                "The environments to remove. ENV can be "
-                "either a name from the venvs shown by vox"
-                "list or the path to an arbitrary venv"
-            ),
-        )
-        subparsers.add_parser("help", help="Show this help message")
+        parser.add_command(self.activate, aliases=["workon", "enter"])
+        parser.add_command(self.deactivate, aliases=["exit"])
+        parser.add_command(self.list, aliases=["ls"])
+        parser.add_command(self.remove, aliases=["rm", "delete", "del"])
+        parser.add_command(self.help)
         return parser
 
-    parser = lazyasd.LazyObject(parser, locals(), "parser")
-
-    aliases = {
-        "create": "new",
-        "workon": "activate",
-        "enter": "activate",
-        "exit": "deactivate",
-        "ls": "list",
-        "rm": "remove",
-        "delete": "remove",
-        "del": "remove",
-    }
-
-    def __init__(self):
-        self.vox = voxapi.Vox()
-
-    def __call__(self, args, stdin=None):
-        """Call the right handler method for a given command."""
-
-        args = self.parser.parse_args(args)
-        cmd = self.aliases.get(args.command, args.command)
-        if cmd is None:
-            self.parser.print_usage()
-        else:
-            getattr(self, "cmd_" + cmd)(args, stdin)
-
-    def cmd_new(self, args, stdin=None):
-        """Create a virtual environment in $VIRTUALENV_HOME with python3's ``venv``."""
+    def new(
+        self,
+        name: Annotated[str, Arg(metavar="ENV")],
```

Comment:
Writing `Annotated[type, Arg(...)`  for each arg seems like a lot of boilerplate...
Can it be switched to `Arg(type, ...)` that returns an `Annotated[...]` object? I think mypy will work with that
(another possible style can be `Arg[type](...)`)

Reply:
1. Arg[] will work here, but we will lose ability to use keyword arguments then
2. I tried Arg() first, with returning type if type_checking is true, but mypy objects anything not in format of `Type[]`